awx_collection/plugins/modules/tower_project_update.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: tower_project_update
+author: "Sean Sullivan (@sean-m-sullivan)"
+short_description: Update a Project in Ansible Tower
+description:
+    - Update a Ansible Tower Project. See
+      U(https://www.ansible.com/tower) for an overview.
+options:
+    name:
+      description:
+        - The name of the project to update.
+      required: True
+      type: str
+      aliases:
+        - project
```

Comment:
In this case it would not because get_one does not attempt to do an ID if it can't find something by name. resolve_name_to_id does allow for a ID and will try an ID after failing to find something by name. But resolve_name_to_id does not allow for additional search criteria. We should think about combining these functionalities.

Reply:
This module now resolves by id or name. I tried to do it as a global change but when coupled with project_update, it would change the name to the ID#, as that project hardcodes where it gets the name from the module args. If we want to implement this collection wide, we would need to rewrite all the modules where this is the case.